database/src/models.rs

```diff
@@ -483,13 +518,33 @@ impl Status {
         match uri {
             Some(x) => Ok(x),
             None => {
-                return Ok(format!(
-                    "{base}/users/{user}/statuses/{id}",
-                    base = BASE_URL.as_str(),
-                    user = self.account(db_conn)?.username,
-                    id = self.id
-                ).into())
+                let account_result = self.account(db_conn);
+                match account_result {
+                    Ok(account) => {
+                        return Ok(format!(
+                            "{base}{path}",
+                            base = BASE_URL.as_str(),
+                            path = self.status_path(&account).unwrap_or_else(|| { "".into() })
+                        ).into())
+                    }
+                    Err(error) => {
+                        return Err(error)
+                    }
+                }
             },
         }
     }
+
+    pub fn status_path<'a>(&'a self, account: &Account) -> Option<Cow<'a, str>> {
+        if account.id == self.account_id {
```

Comment:
I'm not sure if `None` or some sort of error raising functionality is the right path here, but it seemed like some sort of guard was necessary for this factor.